closing handshake failed
Traceback (most recent call last):
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/server.py", line 239, in handler
    await self.close()
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 779, in close
    await asyncio.shield(self.close_connection_task)
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 1309, in close_connection
    self.transport.write_eof()
  File "/Users/aamirsohail/anaconda3/lib/python3.11/asyncio/selector_events.py", line 1111, in write_eof
    self._sock.shutdown(socket.SHUT_WR)
OSError: [Errno 57] Socket is not connected
closing handshake failed
Traceback (most recent call last):
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/server.py", line 239, in handler
    await self.close()
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 779, in close
    await asyncio.shield(self.close_connection_task)
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 1309, in close_connection
    self.transport.write_eof()
  File "/Users/aamirsohail/anaconda3/lib/python3.11/asyncio/selector_events.py", line 1111, in write_eof
    self._sock.shutdown(socket.SHUT_WR)
OSError: [Errno 57] Socket is not connected
closing handshake failed
Traceback (most recent call last):
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/server.py", line 239, in handler
    await self.close()
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 779, in close
    await asyncio.shield(self.close_connection_task)
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 1309, in close_connection
    self.transport.write_eof()
  File "/Users/aamirsohail/anaconda3/lib/python3.11/asyncio/selector_events.py", line 1111, in write_eof
    self._sock.shutdown(socket.SHUT_WR)
OSError: [Errno 57] Socket is not connected
closing handshake failed
Traceback (most recent call last):
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/server.py", line 239, in handler
    await self.close()
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 779, in close
    await asyncio.shield(self.close_connection_task)
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 1309, in close_connection
    self.transport.write_eof()
  File "/Users/aamirsohail/anaconda3/lib/python3.11/asyncio/selector_events.py", line 1111, in write_eof
    self._sock.shutdown(socket.SHUT_WR)
OSError: [Errno 57] Socket is not connected
closing handshake failed
Traceback (most recent call last):
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/server.py", line 239, in handler
    await self.close()
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 779, in close
    await asyncio.shield(self.close_connection_task)
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 1309, in close_connection
    self.transport.write_eof()
  File "/Users/aamirsohail/anaconda3/lib/python3.11/asyncio/selector_events.py", line 1111, in write_eof
    self._sock.shutdown(socket.SHUT_WR)
OSError: [Errno 57] Socket is not connected
closing handshake failed
Traceback (most recent call last):
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/server.py", line 239, in handler
    await self.close()
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 779, in close
    await asyncio.shield(self.close_connection_task)
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 1309, in close_connection
    self.transport.write_eof()
  File "/Users/aamirsohail/anaconda3/lib/python3.11/asyncio/selector_events.py", line 1111, in write_eof
    self._sock.shutdown(socket.SHUT_WR)
OSError: [Errno 57] Socket is not connected
closing handshake failed
Traceback (most recent call last):
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/server.py", line 239, in handler
    await self.close()
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 779, in close
    await asyncio.shield(self.close_connection_task)
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 1309, in close_connection
    self.transport.write_eof()
  File "/Users/aamirsohail/anaconda3/lib/python3.11/asyncio/selector_events.py", line 1111, in write_eof
    self._sock.shutdown(socket.SHUT_WR)
OSError: [Errno 57] Socket is not connected
closing handshake failed
Traceback (most recent call last):
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/server.py", line 239, in handler
    await self.close()
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 779, in close
    await asyncio.shield(self.close_connection_task)
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 1309, in close_connection
    self.transport.write_eof()
  File "/Users/aamirsohail/anaconda3/lib/python3.11/asyncio/selector_events.py", line 1111, in write_eof
    self._sock.shutdown(socket.SHUT_WR)
OSError: [Errno 57] Socket is not connected
closing handshake failed
Traceback (most recent call last):
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/server.py", line 239, in handler
    await self.close()
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 779, in close
    await asyncio.shield(self.close_connection_task)
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 1309, in close_connection
    self.transport.write_eof()
  File "/Users/aamirsohail/anaconda3/lib/python3.11/asyncio/selector_events.py", line 1111, in write_eof
    self._sock.shutdown(socket.SHUT_WR)
OSError: [Errno 57] Socket is not connected
closing handshake failed
Traceback (most recent call last):
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/server.py", line 239, in handler
    await self.close()
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 779, in close
    await asyncio.shield(self.close_connection_task)
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 1309, in close_connection
    self.transport.write_eof()
  File "/Users/aamirsohail/anaconda3/lib/python3.11/asyncio/selector_events.py", line 1111, in write_eof
    self._sock.shutdown(socket.SHUT_WR)
OSError: [Errno 57] Socket is not connected
closing handshake failed
Traceback (most recent call last):
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/server.py", line 239, in handler
    await self.close()
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 779, in close
    await asyncio.shield(self.close_connection_task)
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 1309, in close_connection
    self.transport.write_eof()
  File "/Users/aamirsohail/anaconda3/lib/python3.11/asyncio/selector_events.py", line 1111, in write_eof
    self._sock.shutdown(socket.SHUT_WR)
OSError: [Errno 57] Socket is not connected
closing handshake failed
Traceback (most recent call last):
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/server.py", line 239, in handler
    await self.close()
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 779, in close
    await asyncio.shield(self.close_connection_task)
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 1309, in close_connection
    self.transport.write_eof()
  File "/Users/aamirsohail/anaconda3/lib/python3.11/asyncio/selector_events.py", line 1111, in write_eof
    self._sock.shutdown(socket.SHUT_WR)
OSError: [Errno 57] Socket is not connected
closing handshake failed
Traceback (most recent call last):
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/server.py", line 239, in handler
    await self.close()
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 779, in close
    await asyncio.shield(self.close_connection_task)
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 1309, in close_connection
    self.transport.write_eof()
  File "/Users/aamirsohail/anaconda3/lib/python3.11/asyncio/selector_events.py", line 1111, in write_eof
    self._sock.shutdown(socket.SHUT_WR)
OSError: [Errno 57] Socket is not connected
closing handshake failed
Traceback (most recent call last):
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/server.py", line 239, in handler
    await self.close()
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 779, in close
    await asyncio.shield(self.close_connection_task)
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 1309, in close_connection
    self.transport.write_eof()
  File "/Users/aamirsohail/anaconda3/lib/python3.11/asyncio/selector_events.py", line 1111, in write_eof
    self._sock.shutdown(socket.SHUT_WR)
OSError: [Errno 57] Socket is not connected
closing handshake failed
Traceback (most recent call last):
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/server.py", line 239, in handler
    await self.close()
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 779, in close
    await asyncio.shield(self.close_connection_task)
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 1309, in close_connection
    self.transport.write_eof()
  File "/Users/aamirsohail/anaconda3/lib/python3.11/asyncio/selector_events.py", line 1111, in write_eof
    self._sock.shutdown(socket.SHUT_WR)
OSError: [Errno 57] Socket is not connected
closing handshake failed
Traceback (most recent call last):
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/server.py", line 239, in handler
    await self.close()
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 779, in close
    await asyncio.shield(self.close_connection_task)
  File "/Users/aamirsohail/University of Birmingham - ALPN Laboratory - Research - Projects/online_social_study/otree_projects/social-influence-task-botex/botexenv/lib/python3.11/site-packages/websockets/legacy/protocol.py", line 1309, in close_connection
    self.transport.write_eof()
  File "/Users/aamirsohail/anaconda3/lib/python3.11/asyncio/selector_events.py", line 1111, in write_eof
    self._sock.shutdown(socket.SHUT_WR)
OSError: [Errno 57] Socket is not connected
